ui/pages/confirmations/components/confirm/header/header-info.tsx

```diff
@@ -63,7 +63,10 @@ const HeaderInfo = () => {
       properties: {
         action: 'Confirm Screen',
         location: MetaMetricsEventLocation.SignatureConfirmation,
-        signature_type: currentConfirmation?.type,
+        signature_type:
+          currentConfirmation && 'msgParams' in currentConfirmation
+            ? currentConfirmation.msgParams?.signatureMethod
+            : currentConfirmation?.type,
```

Comment:
to save storage space, I'd recommend omitting `signature_type` when the confirmation is not a signature

actually, since we're expanding to non-signature transactions, I think we can check if it's a signature, then provide `signature_type`, else we can provide `transaction_type`.  

We can find the available properties for the specific event here:
https://www.notion.so/11519916349f4263b520940150fa0f7b?v=63c68b587844461d899f50c6aaae7215&p=3deec398d4784bcea00b012cdf1cde03&pm=s
I added the `signature_type` and `transaction_type` properties since we expanded this event to support the Transactions category for the redesign

Reply:
Makes sense, will change that, Just noticed two small things on the event
-  `category: MetaMetricsEventCategory.Transactions` this is set regardless of whether it is a transaction or signature, should we change this to add a new category called Confirmations? Or based on the type decide if its a Singature or Transaction and use that as a category?
- same for `properties.location` right now it is set to `MetaMetricsEventLocation.SignatureConfirmation`, this should also change based on type right?

Reply:
re: category 
That's interesting proposition. @bschorchit would this be helpful?

re: properties.location
Great callout! I agree this should be updated based on type

Reply:
From a cleanup perspective, it does make sense to me to have a Signature and a Transaction category. I wouldn't group it all under Confirmations as some of the Transactions events go beyond confirmations scope and are relevant for transactions team too for example. 
Right now, though, we don't leverage category in a meaningful way within Mixpanel. So, we can also leave this small improvement for the future. 

For Signature Requested event on Mixpanel, I currently see `inpage_provider` as the category. 
<img width="380" alt="Screenshot 2024-06-13 at 17 18 39" src="https://github.com/MetaMask/metamask-extension/assets/34968356/6fdf0045-e9c0-42cf-b8c6-4f69d9267265">

Reply:
great callout too @bschorchit! In the code, where Signature <Requested,Approved,Rejected>, is tracked, all of the events here use `inpage_provider` as the category. Should we update the category here as well? 

file: app/scripts/lib/createRPCMethodTrackingMiddleware.js

Reply:
This distinction is preferred by the data team to favor easier readability for someone with possibly less context reading or interacting with a dashboard.